app/features/atp/fraud-check/ui/idx/controller.ts

```diff
@@ -0,0 +1,53 @@
+import { Request, Response, Router } from "express";
+import { PageSetup } from "../../../../../interfaces/PageSetup";
+import { pathName } from "../../../../../paths";
+import Logger from "../../../../../utils/logger";
+
+const template = "atp/fraud-check/ui/idx/view.njk";
+const logger: Logger = new Logger();
+
+interface FraudCheck {
+  fraudCheckData: any;
+}
+
+const getFraudCheckPage = (req: Request, res: Response): void => {
+  return res.render(template, { language: req.i18n.language });
+};
+
+const getJsonFromString = (data: string): Record<string, any> => {
+  try {
+    return JSON.parse(data);
+  } catch (e) {
+    logger.warn(`Failed to parse json data: ${e}`, "failed-parsing-json");
+    return {};
+  }
+};
+
+const postFraudCheck = async (req: Request, res: Response): Promise<void> => {
+  const fraudCheckData = getJsonFromString(req.body["activityHistoryData"]);
+
+  const fraudCheck: FraudCheck = {
+    fraudCheckData: fraudCheckData,
+  };
+
+  req.session.sessionData.fraudChecks =
+    req.session.sessionData.fraudChecks || [];
+  req.session.sessionData.fraudChecks.push(fraudCheck);
```

Comment:
This is the same logic we had for the `identityEvidence`, but unfortunately it's got a bug in it!

What happens if you re-edit this form? With this fraud check there would now be two entries in place.

What we'll need to do is to find and remove the old entry, and add a new one. If we don't care about the order of these kinds of data, then we can `push` it onto the end, otherwise we'll need some sort of `splice`ing logic.

Reply:
This will be changed in as part of PYI-186 overarching ticket.